pos_partner_firstname/static/src/js/ClientDetailsEdit.js

```diff
@@ -0,0 +1,93 @@
+odoo.define("pos_partner_firstname.ClientDetailsEdit", function (require) {
+    "use strict";
+
+    const {_t} = require("web.core");
+    const ClientDetailsEdit = require("point_of_sale.ClientDetailsEdit");
+    const Registries = require("point_of_sale.Registries");
+
+    const PosClientDetailsEdit = (ClientDetailsEdit) =>
+        class extends ClientDetailsEdit {
+            constructor() {
+                super(...arguments);
+                var self = this;
+                this.partner_names_order = "last_first";
+                this.rpc({
+                    model: "res.partner",
+                    method: "get_names_order",
+                    args: [],
+                }).then(function (partner_names_order) {
+                    if (partner_names_order) {
+                        self.partner_names_order = partner_names_order;
+                    }
+                });
+            }
+            _update_client_name(checked) {
+                if (!checked) {
+                    var lastname = this.props.partner.lastname
+                        ? this.props.partner.lastname
+                        : "";
+                    var firstname = this.props.partner.firstname
+                        ? this.props.partner.firstname
+                        : "";
+                    var name = null;
+                    if (this.partner_names_order === "last_first_comma") {
+                        name = lastname + ", " + firstname;
+                    } else if (this.partner_names_order === "first_last") {
+                        name = firstname + " " + lastname;
+                    } else {
+                        name = lastname + " " + firstname;
+                    }
+                    this.props.partner.name = name.trim();
+                }
+            }
+            saveChanges() {
+                const processedChanges = {};
+                for (const [key, value] of Object.entries(this.changes)) {
+                    if (this.intFields.includes(key)) {
+                        processedChanges[key] = parseInt(value) || false;
+                    } else {
+                        processedChanges[key] = value;
+                    }
+                }
+                var checked = $(".is_company").is(":checked");
+                if (!checked) {
+                    if (
+                        (!this.props.partner.firstname &&
+                            !processedChanges.firstname) ||
+                        processedChanges.firstname === "" ||
+                        (!this.props.partner.lastname && !processedChanges.lastname) ||
+                        processedChanges.lastname === ""
+                    ) {
+                        return this.showPopup("ErrorPopup", {
+                            title: _t("Both Customer First and Last Name Are Required"),
```

Comment:
In back office, the name or the surname is mandatory.

In front office, it is not the same.

Should be great to be similar.

![image](https://user-images.githubusercontent.com/3407482/161525847-79ebfd11-d52f-4a3a-9740-c16fd8a39c3c.png)

Reply:
Agree